finality-aleph/src/sync/data.rs

```diff
@@ -180,34 +218,39 @@ impl<J: Justification, N: GossipNetwork<VersionedNetworkData<J>>> VersionWrapper
 }
 
 #[async_trait::async_trait]
-impl<J: Justification, N: GossipNetwork<VersionedNetworkData<J>>> GossipNetwork<NetworkData<J>>
-    for VersionWrapper<J, N>
+impl<B: Block, J: Justification, N: GossipNetwork<VersionedNetworkData<B, J>>>
+    GossipNetwork<NetworkData<B, J>> for VersionWrapper<B, J, N>
 {
     type Error = N::Error;
     type PeerId = N::PeerId;
 
-    fn send_to(&mut self, data: NetworkData<J>, peer_id: Self::PeerId) -> Result<(), Self::Error> {
-        self.inner.send_to(VersionedNetworkData::V1(data), peer_id)
+    fn send_to(
+        &mut self,
+        data: NetworkData<B, J>,
+        peer_id: Self::PeerId,
+    ) -> Result<(), Self::Error> {
+        self.inner.send_to(VersionedNetworkData::V2(data), peer_id)
```

Comment:
Here and in the other sends we should actually be sending both versions, for backwards compatibility. This requires a `(Try?)From` implementation in the other direction.

I think in at least some places this will double the bandwidth used, but oh well.